fuel-merkle/src/common/position_path.rs

```diff
@@ -47,9 +50,15 @@ pub struct PositionPathIter {
 }
 
 impl PositionPathIter {
+    /// Panics if leaves_count is zero, as the tree is not valid
     pub fn new(root: Position, leaf: Position, leaves_count: u64) -> Self {
         Self {
-            rightmost_position: Position::from_leaf_index(leaves_count - 1),
+            rightmost_position: Position::from_leaf_index(
+                leaves_count
+                    .checked_sub(1)
+                    .expect("Path to a tree without leaves"),
+            )
+            .unwrap(),
```

Comment:
I think we can't guaranty that this will not fail. Is someone provides invalid `leaves_count`, it will fail

Reply:
If the tree has a root, then `leaves_count` will be at least 1. In real use cases, this should be okay. But if someone were to provide arguments that are inconsistent with a real tree, this could fail. I don't think that's an issue.